Authentication and QoS feature

ashish-vmware opened the pull request:
- Added username password setup methods
- Added QoS configuration validations and setup methods
- Added QoS conf class which does the abstraction for QoS related configurations
- Added protocol utility class which will encapsulate result codes for various protocols

Commit:
TLS Feature:
- Adding TLS support and certificate validations in current MQTT implementation
- Adding identity layer to encapsulate gateway identity and TLS configurations

Commit:
Updates in TLS implementations
- Making keepalive parameter configurable
- Updating client certification validation method
- Adding sample MQTT configurations

Commit:
- Changes in certificate validations

Commit:
Features : Username/Password based authentication and QoS Support
- Added username password setup methods
- Added QoS configuration validations and setup methods
- Added QoS conf class which does the abstraction for QoS related configurations
- Added protocol utility class which will encapsulate result codes for various protocols

diwanc commented on liota/transports/mqtt.py:
```diff
@@ -44,29 +46,112 @@ def on_subscribe(self, client, userdata, mid, granted_qos):
 
     def on_message(self, client, userdata, msg):
         log.debug("On Message {0} {1} {2}".format(msg.topic, str(msg.qos), str(msg.payload)))
+        self.client.disconnect()
```
What does this line signify? self.client.disconnect()

diwanc commented on liota/transports/mqtt.py:
```diff
@@ -44,29 +46,112 @@ def on_subscribe(self, client, userdata, mid, granted_qos):
+        else:
+            log.debug("Connected with result code " + str(rc) + " : " + self.mqttresults.resultcodes.get(rc))
+
+    def __init__(self, gw_identity, TLSConf, QoSConf, url, port, keepalive, tls_support=True, enable_authentication = True):
```
1) Rename QoSConf to qos_details
2) Rename TLSConf to tls_details
3) Get rid of tls_support boolean parameter. Check for tls_details presence instead
4) Ensure default values are defined for qos attributes. Check if the paho client implementation provides default values, in which case we should be good

diwanc commented on liota/transports/mqtt.py:
```diff
@@ -44,29 +46,112 @@ def on_subscribe(self, client, userdata, mid, granted_qos):
+                self.client.username_pw_set(self.gw_identity.username, self.gw_identity.password)
+
+        # Set QoS parameters
+        if self.QoSConf.qoslevel:
```
What if the qoslevel is not present?

diwanc commented on liota/transports/mqtt.py:
```diff
@@ -44,29 +46,112 @@ def on_subscribe(self, client, userdata, mid, granted_qos):
+
 
+        # Connect with MQTT Broker
+        self.client.on_connect = self.on_connect
```
Move on_connect binding to the init method where other callback methods are registered

diwanc commented on liota/utilities/protocol_utility.py:
```diff
@@ -0,0 +1,43 @@
+# -*- coding: utf-8 -*-
```
This utility is not required.
Instead leverage paho provided error code and conn ack code lookup methods. We should not attempt to re-map code and description once again. Refer https://github.com/eclipse/paho.mqtt.python/blob/master/src/paho/mqtt/client.py

diwanc commented on liota/identity/qos_conf.py:
```diff
@@ -0,0 +1,39 @@
+# -*- coding: utf-8 -*-
```
1) Rename QoSConf to QoSDetails
2) This file cannot be in the identity layer. Move the class to mqtt.py

diwanc commented on liota/identity/tls_conf.py:
```diff
@@ -0,0 +1,43 @@
+log = logging.getLogger(__name__)
+
+# Encapsulates TLS configurations
+class TLSConf:
```
Rename to TLSDetails

Commit:
Changes
- Renamed TLS and QoS related classes and variables
- Removed tls_support boolean, now it'll check for TLS configurations are defined or not
- Moved on_connect binding to init method
- Removed qos_conf from identity layer and protocol_utility from utilities

Commit:
no message

Commit:
Merge branch 'master' of https://github.com/ashish-vmware/liota into Authentication_QoS_feature

Conflicts:
	example/sampleProp.conf

Commit:
Merge branch 'master' of https://github.com/diwanc/liota

Commit:
Merge branch 'master' into Authentication_QoS_feature

Commit:
Handling return values and errors : Paho publish and subscribe methods

Commit:
Revert "Handling return values and errors : Paho publish and subscribe methods"

This reverts commit c4e16bc811f46a86b08893cff7684cb8ac25ac1e.
